test/catalog/catalog_sql_test.cpp

```diff
@@ -0,0 +1,75 @@
+#include <memory>
+#include <pqxx/pqxx>  // NOLINT
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "common/settings.h"
+#include "gtest/gtest.h"
+#include "main/db_main.h"
+#include "test_util/test_harness.h"
+
+namespace noisepage::catalog {
+
+// These are end-to-end tests of the catalog which cannot be performed with the
+// JUnit trace framework because we expect divergent behavior from PostgreSQL
```

Comment:
AFAK, the trace framework is comparing the hash value of the execution results. I don't understand how divergent behavior from PostgresSQL prevents us from using the JUnit trace framework. Could you explain it?

Reply:
The JUnit framework generates the hashes and result sets in the traces from a run against PostgreSQL which we are storing and then using as a reference point for what Noisepage should be outputting.  We could hand-craft a trace file, but I think that is fragile in the long run and not substantially different from what we are doing here.